### PR TITLE
Feature/not found page

### LINKED ISSUE #8: Provide not-found page
Mockups ready

### PR DESCRIPTION
**Pull request by patrykbuzowiczsabre**
fixes #8 and #5

### CODE REVIEW

File: src/components/errors/NotFound.tsx [hunk 1]
```diff
@@ -1,11 +1,30 @@
+import {History} from 'history';
 import * as React from 'react';
-import Error from './Error';
+import {InjectedIntlProps, injectIntl} from 'react-intl';
+import {withRouter} from 'react-router';
```
Comment (wojciech-kumon) [inline src/components/errors/NotFound.tsx]: use `react-router-dom` - `react-router` is not present in `package.json`

File: src/components/errors/NotFound.tsx [hunk 2]
```diff
@@ -1,11 +1,30 @@
...
+import NotFoundComponent from './NotFoundComponent';
 
-const NotFound = () => (
-  <VerticalCenterContainer>
-    <Error titleKey="pageNotFound.title" textKey="pageNotFound.text" />
-  </VerticalCenterContainer>
-);
+export class NotFound extends React.PureComponent<NotFoundProps & InjectedIntlProps> {
+  render() {
+    const {intl: {formatMessage}, history} = this.props;
```
Comment (wojciech-kumon) [inline src/components/errors/NotFound.tsx]: unused `formatMessage` (and `intl`) - remove `injectIntl` wrapper

File: src/components/errors/NotFoundComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+interface NotFoundComponentProps {
+  titleKey: string;
+  messageKey: string;
+  buttonKey: string;
+  onButtonClick(): void;
+}
+
+export const HighWideButton = styled(HighButton)`
```
Comment (wojciech-kumon) [inline src/components/errors/NotFoundComponent.tsx]: when adding styles to existing styled-component use `HighButton.extend` method

File: src/components/common/styledComponents.tsx [hunk 4]
```diff
@@ -22,3 +24,8 @@ export const MainContainerBody = styled.div`
 export const VerticalCenterContainer = MainContainerBody.extend`
   justify-content: center;
 `;
+
+export const Background = styled.div`
```
Comment (wojciech-kumon) [inline src/components/common/styledComponents.tsx]: reuse this component in `LoginPage.tsx` (WhiteForm)

File: src/components/proposalFinder/FinderWindow.tsx [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
+import {Background} from '../common/styledComponents';
 
 export class FinderWindow extends React.PureComponent<FinderWindowProps> {
   render() {
     const {formatMessage, findTripProposal} = this.props;
     return (
       <Background className="mx-3 p-4">
-        <TravelerPortalHeader/>
-        <FinderForm formatMessage={formatMessage} onSubmit={findTripProposal}/>
+        <TravelerPortalHeader />
```
Comment (wojciech-kumon) [inline src/components/proposalFinder/FinderWindow.tsx]: Are this whitespace changes needed?

File: src/__tests__/reducers/viewProposalReducer.test.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import {viewProposalReducer, ViewProposalState} from '../../reducers/viewProposalReducer';
+import {viewProposalReducer, ViewProposalState, ErrorType} from '../../reducers/viewProposalReducer';
 import {viewProposalStart, viewProposalSuccess, viewProposalError} from '../../actions/viewProposal/viewActions';
 import {VIEW_PROPOSAL_START, ProposalData, ViewProposalSuccessAction} from '../../actions/viewProposal/viewActionTypes';
```
Comment (wojciech-kumon) [inline src/__tests__/reducers/viewProposalReducer.test.ts]: `VIEW_PROPOSAL_START`, `ViewProposalSuccessAction` unused

File: src/components/footer/Footer.tsx [hunk 7]
```diff
@@ -1,5 +1,7 @@
 import * as React from 'react';
 import styled from 'styled-components';
+import {footer} from '../../style/colors';
+import {mainPadding} from '../../style/sizes';
 
 class Footer extends React.Component<{}, FooterState> {
   constructor(props) {
```
Comment (wojciech-kumon) [inline src/components/footer/Footer.tsx]: Switch this component to PureComponent please (move `currentYear` from state to class field)

File: src/reducers/viewProposalReducer.ts [hunk 8]
```diff
@@ -13,21 +13,47 @@ export interface ViewProposalState {
...
+    const errorType = isNotFoundError(action.error) ? ErrorType.notFound : ErrorType.unknown;
+    return {
+      ...state,
+      isFetching: false,
+      error: { data: action.error, type: errorType }
+    };
   }
 });
+
+const isNotFoundError = (data: any) => {
```
Comment (wojciech-kumon) [inline src/reducers/viewProposalReducer.ts]: remove braces and `return`

File: src/components/errors/NotFound.tsx [hunk 9]
```diff
@@ -1,11 +1,30 @@
...
+    const navigateToHome = () => history.push('/');
 
-export default NotFound;
+    return (
+      <VerticalCenterContainer>
+        <NotFoundComponent
+          titleKey="genericError.title"
```
Comment (wojciech-kumon) [inline src/components/errors/NotFound.tsx]: translations are not matching mocks

File: src/components/errors/NotFoundComponent.tsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+export class NotFoundComponent extends React.PureComponent<NotFoundComponentProps & InjectedIntlProps> {
+  render() {
+    const { titleKey, messageKey, buttonKey, onButtonClick, intl: { formatMessage } } = this.props;
+
+    return (
+      <div>
+        <MaxWidthDiv maxWidth={30} unit="rem" className="mx-auto text-center">
+          <Background className="p-4">
+            <h2>{formatMessage({id: titleKey})}</h2>
+            <p>{formatMessage({id: messageKey})}</p>
```
Comment (wojciech-kumon) [inline src/components/errors/NotFoundComponent.tsx]: description should be bolded

File: src/components/proposalFinder/FinderWindow.tsx (hunk 5 above)
Comment (patrykbuzowiczsabre) [reply src/components/proposalFinder/FinderWindow.tsx]: no, it's just how vscode automatically formats JSX. I did some digging and found nothing about how to change this

File: src/components/errors/NotFound.tsx (hunk 9 above)
Comment (patrykbuzowiczsabre) [reply src/components/errors/NotFound.tsx]: let me ask Adam if we can have them a little more descriptive and then adjust if needed

File: src/components/errors/NotFound.tsx [hunk 11]
```diff
@@ -1,11 +1,30 @@
...
 
-const NotFound = () => (
-  <VerticalCenterContainer>
-    <Error titleKey="pageNotFound.title" textKey="pageNotFound.text" />
-  </VerticalCenterContainer>
-);
+export class NotFound extends React.PureComponent<NotFoundProps & InjectedIntlProps> {
+  render() {
+    const {intl: {formatMessage}, history} = this.props;
+    const navigateToHome = () => history.push('/');
```
[the remark this answers is not in the document]
Comment (patrykbuzowiczsabre) [reply src/components/errors/NotFound.tsx]: fixed

File: src/components/errors/NotFound.tsx (hunk 1 above)
Comment (patrykbuzowiczsabre) [reply src/components/errors/NotFound.tsx]: right

File: src/components/errors/NotFound.tsx (hunk 2 above)
Comment (patrykbuzowiczsabre) [reply src/components/errors/NotFound.tsx]: done

**Pull request merged**